src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -556,13 +556,14 @@ private BoundExpression MakeImplicitConversion(BoundExpression rewrittenOperand,
 
                 TypeSymbol userDefinedConversionRewrittenType = conversion.Method.ReturnType;
 
-                if (rewrittenOperand.Type != conversion.Method.ParameterTypes[0])
+                // the conversion only needs to happen for non-nullable valuetypes
+                if (rewrittenOperand.Type != conversion.Method.ParameterTypes[0] && !userDefinedConversionRewrittenType.CanBeAssignedNull())
```

Comment:
The lifting conversion here attempts to make the result nullable (either nullable valuetype or reference). `rewrittenOperand.Type` is always `Nullable<T>` here. But for surity, I will write test cases for the reverse case.